src/Components/Web/src/Routing/NavLink.cs

```diff
@@ -106,7 +106,12 @@ private void OnLocationChanged(object? sender, LocationChangedEventArgs args)
         }
     }
 
-    private bool ShouldMatch(string currentUriAbsolute)
+    /// <summary>
+    /// Determines whether the current URI should match the link.
+    /// </summary>
+    /// <param name="currentUriAbsolute">The absolute URI of the current location.</param>
+    /// <returns>True if the link should be highlighted as active; otherwise, false.</returns>
+    protected virtual bool ShouldMatch(string currentUriAbsolute)
```

Comment:
Is there a reason for the new public API bit? If it's only for testing, then you should be able to do `private protected` with `InternalsVisibleTo` in the test assembly (which should already have this).

In addition to that, I think we want to fix the behavior in the base class without requiring users to subclass the component.

Reply:
The reason is: fixing the original issue. I would like the users to have a way to override this method in their projects, not only in our tests.